ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -227,9 +93,20 @@ const ApproveInfo = () => {
     <>
       <ApproveStaticSimulation />
       <ApproveDetails />
-      {!isNFT && <SpendingCap />}
+      {!isNFT && (
+        <SpendingCap
+          setIsOpenEditSpendingCapModal={setIsOpenEditSpendingCapModal}
+          customSpendingCap={customSpendingCap}
+        />
+      )}
       <GasFeesSection />
       {showAdvancedDetails && <AdvancedDetails />}
+      <EditSpendingCapModal
+        isOpenEditSpendingCapModal={isOpenEditSpendingCapModal}
+        setIsOpenEditSpendingCapModal={setIsOpenEditSpendingCapModal}
+        customSpendingCap={customSpendingCap}
+        setCustomSpendingCap={setCustomSpendingCapCandidate}
```

Comment:
Rather than adding further logic and complexity here, could we empower the modal to encapsulate the update logic and update the actual transaction?

Reply:
The issue is that the spending cap needs to be shown in the `<SpendingCap />` component above it, so a variable needs to exist upstream from the modal component, in this <Approve /> section.